packages/mask/src/plugins/Referral/helpers/buffer.ts

```diff
@@ -0,0 +1,152 @@
+            table[i16 + j] = alphabet[i] + alphabet[j]
+        }
+    }
+    return table
+})()
+
+function hexSlice(buf: Uint8Array, start?: number, end?: number) {
+    const len = buf.length
+
+    if (!start || start < 0) start = 0
+    if (!end || end < 0 || end > len) end = len
+
+    let out = ''
+    for (let i = start; i < end; i += 1) {
+        out += hexSliceLookupTable[buf[i]]
+    }
+    return out
+}
+
+export function bufToHexString(buffer: Uint8Array, start?: number, end?: number) {
+    // No need to verify that "this.length <= MAX_UINT32" since it's a read-only
+    // property of a typed array.
+
+    // This behaves neither like String nor Uint8Array in that we set start/end
+    // to their upper/lower bounds if the value passed is out of range.
+    // undefined is handled specially as per ECMA-262 6th Edition,
+    // Section 13.3.3.7 Runtime Semantics: KeyedBindingInitialization.
+    if (start === undefined || start < 0) {
+        start = 0
+    }
+    // Return early if start > this.length. Done here to prevent potential uint32
+    // coercion fail below.
+    if (start > buffer.length) {
+        return ''
+    }
+
+    if (end === undefined || end > buffer.length) {
+        end = buffer.length
+    }
+
+    if (end <= 0) {
+        return ''
+    }
+
+    // Force coercion to uint32. This will also coerce false/NaN values to 0.
+    // eslint-disable-next-line no-bitwise
+    end >>>= 0
+    // eslint-disable-next-line no-bitwise
+    start >>>= 0
+
+    if (end <= start) {
+        return ''
+    }
+
+    return hexSlice(buffer, start, end)
+}
+
+function buf(b: Bytes | Uint8Array | HexTypes | number | BigIntTypes | BigNumber): Uint8Array {
+    if (b === null || b instanceof Uint8Array) return b
+
+    if (typeof b === 'number') {
+        return hexToArrayBuffer(new BigNumber(b).toString(16).slice(2))
+    }
+
+    if (typeof b === 'bigint') return hexToArrayBuffer(toEvenLength(b.toString(16)))
+
+    if (typeof b === 'string') {
+        if (!b.startsWith('0x')) throw new Error('unsupported')
+        const hex = b.slice(2)
+        if (hex.length % 2 !== 0 || !isHexString(b)) {
+            throw new Error('invalid hex string')
+        }
+
+        return hexToArrayBuffer(hex)
+    }
+
+    // This should return the most compact buffer version without leading zeros, so safe for RLP encodings.
+    if (b instanceof BigNumber || (b as any)?._isBigNumber === true)
+        return hexToArrayBuffer((b as any).toHexString().slice(2))
+
+    throw new Error('unsupported: ')
+}
+export function toBigInt(b: BigIntTypes | Uint8Array | HexTypes | number | BigNumber): bigint {
+    if (typeof b === 'bigint') return b
+    if (typeof b === 'number') return BigInt(b)
+    if (b instanceof Uint8Array) return toBigInt(buf(b))
+    if (typeof b === 'string' && b.startsWith('0x')) return BigInt(b)
+    if (b instanceof BigNumber || (b as any)?._isBigNumber === true) return BigInt(b as any)
```

Comment:
```suggestion
    if (b instanceof BigNumber || (b as any)?._isBigNumber === true) return BigInt(b as BigNumber)
```

Reply:
Do you have an idea how to resolve this typescript issue?
<img width="710" alt="Screenshot 2022-06-03 at 11 28 26" src="https://user-images.githubusercontent.com/32291396/171837304-3220b811-af04-4c18-a262-c9ea16b5ff20.png">